spec/timezone.html

```diff
@@ -30,73 +30,76 @@ <h1>Time Zone Names</h1>
       </p>
-
       <emu-alg>
-        1. Assert: Type(_timeZone_) is String.
-        1. Let _tzText_ be ! StringToCodePoints(_timeZone_).
-        1. Let _tzUpperText_ be the result of toUppercase(_tzText_), according to the Unicode Default Case Conversion algorithm.
-        1. Let _tzUpper_ be ! CodePointsToString(_tzUpperText_).
-        1. If _tzUpper_ and *"UTC"* are the same sequence of code points, return *true*.
+        1. If _timeZone_ is an ASCII-case-insensitive match for *"UTC"*, return *true*.
+        1. If the implementation-defined built-in time zones include a named time zone whose name is an ASCII-case-insensitive match for _timeZone_, return *true*.
         1. Return *false*.
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-canonicalizetimezonename" aoid="CanonicalizeTimeZoneName">
-      <h1>CanonicalizeTimeZoneName ( _timeZone_ )</h1>
+    <emu-clause id="sec-canonicalizetimezonename" type="abstract operation">
+      <h1>
+        CanonicalizeTimeZoneName (
+          _timeZone_: a String value that is a valid time zone name as verified by IsValidTimeZoneName,
+        )
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It returns the canonical and case-regularized form of _timeZone_.</dd>
+      </dl>
 
       <p>
         An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the CanonicalizeTimeZoneName abstract operation as specified in the ECMA-402 specification.
         If an ECMAScript implementation does not include the ECMA-402 API the following specification of the CanonicalizeTimeZoneName abstract operation is used.
       </p>
+
       <p>
-        The abstract operation CanonicalizeTimeZoneName takes argument _timeZone_, a String value, and returns the canonical and case-regularized form of _timeZone_.
-      </p>
-      <p>
-        CanonicalizeTimeZoneName may only be called with a _timeZone_ parameter that is a valid time zone name according to IsValidTimeZoneName.
-      </p>
-      <p>
-        The minimum implementation of CanonicalizeTimeZoneName, supporting only the *"UTC"* time zone, performs the following steps when called:
+        In the descriptions below, String value _A_ is an ASCII-case-insensitive match for String value _B_ if the String value derived from _A_ by replacing each occurrence of a lowercase ASCII letter code unit (0x0061 through 0x007A, inclusive) with the corresponding uppercase ASCII letter code unit (0x0041 through 0x005A, inclusive) while preserving all other code units is exactly the same sequence of code units as the String value that is derived from _B_ in the same way.
```

Comment:
There's admittedly a lot of duplication here. What would you think of removing IsValidTimeZoneName from both ECMA-262 and ECMA-402 by merging it into CanonicalizeTimeZoneName (which every invocation is already followed by) such that the latter would return undefined when given an invalid time zone name rather than requiring that the name has already been validated (and if there's approval, would that be in scope or out of scope for this PR)?

Reply:
I think we should consider that out of scope for this PR, but it seems like a good idea.